extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -94,14 +100,84 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
   }
 
   /**
-   * It triggers the resize on the rendering engine.
+   * It triggers the resize on the rendering engine, and renders the viewports
+   *
+   * @param isGridResize - if the resize is triggered by a grid resize
+   * this is used to avoid double resize of the viewports since if the
+   * grid is resized, all viewports will be resized so there is no need
+   * to resize them individually which will get triggered by their
+   * individual resize observers
    */
-  public resize() {
-    const immediate = true;
-    const keepCamera = true;
+  public resize(isGridResize = false) {
+    // if there is a grid resize happening, it means the viewport grid
+    // has been manipulated (e.g., panels closed, added, etc.) and we need
+    // to resize all viewports, so we will add a timeout here to make sure
+    // we don't double resize the viewports when viewports in the grid are
+    // resized individually
+    if (isGridResize) {
+      this.performResize();
+      this.resetGridResizeTimeout();
+      this.resizeQueue = [];
+      clearTimeout(this.viewportResizeTimer);
+    } else {
+      this.enqueueViewportResizeRequest();
+    }
+  }
+
+  private enqueueViewportResizeRequest() {
+    this.resizeQueue.push(false); // false indicates viewport resize
+
+    clearTimeout(this.viewportResizeTimer);
+    this.viewportResizeTimer = setTimeout(() => {
+      this.processViewportResizeQueue();
+    }, this.gridResizeDelay);
+  }
+
+  private processViewportResizeQueue() {
+    const isGridResizeInQueue = this.resizeQueue.some(isGridResize => isGridResize);
+    if (this.resizeQueue.length > 0 && !isGridResizeInQueue && !this.gridResizeTimeOut) {
+      this.performResize();
+    }
+
+    // Clear the queue after processing viewport resizes
+    this.resizeQueue = [];
+  }
+
+  private performResize() {
+    const isImmediate = false;
+
+    const viewports = this.getRenderingEngine().getViewports();
+
+    // Store the zoom scale before resizing.
+    viewports.forEach(viewport => {
+      const zoomLevel = viewport.getZoom();
```

Comment:
Could you instead break up getPresentation into two child functions (or maybe more eventually)  getPositionPresentation and getLutPresentation, and then use getPositionPresentation to fetch:
zoom,
pan,
viewUp,
viewNormal,

and then use the set on that to restore the position presentation to a viewport after resize?
the pan values will have to be normalized to the total canvas size rather than being in canvas units
That will give consistent handling with restoring a viewport after it hasn't been displayed, or using a viewport to display another display set.  
That way as we improve the display positioning memory, it will get applied repeatedly in teh same way.

I think this is a better solution than remembering the camera itself

Reply:
Done